### PR TITLE
Implement top blogs section

### LINKED ISSUE #80: Implement "Top Blogs" Section on Home Page
Create a "Top Blogs" section (below the "Places" section) on the Home page which displays the top blogs based on Likes/Dislikes ratio (Take care of 0/0 case). The section should include a grid layout for displaying blog cards.
Each blog card should include the following information:
- Title of the blog
- Thumbnail image
- Author's email
- The number of Likes and Dislikes.
- A link to the full blog post (Or create a modal similar to Place in the Places section)

The "Top Blogs" section should display a reasonable number of blogs (2 rows) to avoid overwhelming the user. The "Top Blogs" section should be responsive and adapt to various screen sizes

Note:

- Create a new branch for every task to isolate changes
- Start working on the issue only when it is assigned to you
- Submit the PR on time, no extra time will be allotted

### PR DESCRIPTION
**Pull request by Nandika-A**
## Description

Selects top 6 blogs based on like/dislike and displays them in 2 rows on homepage.
## Related Issue

Fixes #80 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] I have registered myself at [Contrihub website](https://sac.mnnit.ac.in/contrihub).
- [x] My code follows the code style of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

### CODE REVIEW

File: home/views.py [hunk 1]
```diff
@@ -166,4 +170,15 @@ def get_weather_data(latitude,longitude):
...
+def get_top_posts():
+    top_posts = []
+    all_posts = Post.objects.all() 
+    for post in all_posts:
+        if post.likes > 0 or post.dislikes > 0:
+            top_posts.append(post)
+    try:
+        top_posts.sort(key=lambda x: x.likes/x.dislikes, reverse=True)
+    except:
+        top_posts.sort(key=lambda x: x.likes, reverse=True)
```
Comment (kunal2812) [inline home/views.py:182]: Consider those blogs for "Top Blogs" too where #Likes and #Dislikes are both 0 as the section is empty even though blogs are present.

**Comment by Nandika-A** [comment]
@kunal2812 Sir I have included that case. I have taken like/dislike as (like-dislike)/dislike, otherwise, it would give the same result for 0/1 and 0/0.

File: home/views.py (hunk 1 above)
Comment (Nandika-A) [reply home/views.py:181]: @kunal2812 sir can you check the changes?

**Review by kunal2812 — COMMENTED** [review]
image

According to your logic the correct order should be Blog2 > Blog3 > Blog1. But your code is not working as expected because 0 likes and 0 dislikes is obviously better than 0 likes and 1 dislike.

**Comment by Nandika-A** [comment]
@kunal2812 I have corrected this now.

**Review by kunal2812 — APPROVED** [review]

**Pull request merged**